### PR TITLE
Global Notices: new design and cleanup

### PR DESCRIPTION
**Pull request by mtias**
Picking up the work for better global notices.

Reference:
screen shot 2016-01-28 at 3 47 15 pm

screen shot 2016-01-28 at 3 48 13 pm

cc @rickybanister feel free to pick it up from here and tweak the styles until we are comfortable.

### CODE REVIEW

**Comment by rickybanister** [comment]
I made some further tweaks to this. 

They now look like:

screen shot 2016-01-14 at 3 23 50 pm

screen shot 2016-01-14 at 3 24 07 pm

screen shot 2016-01-14 at 3 24 28 pm

TODO:
Need to add a mechanism for adding padding to the main content body for the narrowest breakpoint to prevent the fixed position notice from covering content. Perhaps a body-class gets added when notices are present? Or a ghost element adjacent to the notice to create space?

**Comment by mtias** [comment]
> Or a ghost element adjacent to the notice to create space?

That's what we had in older notices, I'd like us to avoid it since it was a bit cumbersome.

**Comment by rickybanister** [comment]
I most recently pushed a change that puts mobile global notices stuck to the bottom of the screen. Let's try that out a bit and see if we hate it.

Looks like this:

screen shot 2016-01-14 at 4 56 17 pm

It may be blasphemous, but Material guidelines put toasts/snackbars at the bottom and the lack of interruptions to the page flow are really nice.

**Comment by mtias** [comment]
Changing the text to be:

image

**Comment by mtias** [comment]
The plugins ones are a bit long:

image

**Comment by mtias** [comment]
Disconnected message looks like this:

image

**Comment by mtias** [comment]
@rickybanister what do we need to keep moving here?

**Comment by rickybanister** [comment]
We still need to address the following:

- [ ] make the global notice z-index go behind the notifications tab
- [ ] shorten verbose strings
- [ ] lots of testing
- [ ] buy-in for the mobile-width bottom placement

**Comment by mtias** [comment]
> shorten verbose strings

Is this required or a good to have?

**Comment by rickybanister** [comment]
I was just following your lead on that...I think we could do that in follow-up PRs.

**Comment by rickybanister** [comment]
also we should probably 

- [ ] remove `/client/notices`

**Comment by rickybanister** [comment]
I fixed the z-index issue, registered our z-index properly, and prevented notices from wrapping. 

Before:
screen shot 2016-01-28 at 3 20 00 pm

After:
screen shot 2016-01-28 at 3 38 17 pm

h/t @gwwar for the z-index registration help

**Comment by drw158** [comment]
link to the Material guidelines mentioned above: https://www.google.com/design/spec/components/snackbars-toasts.html#snackbars-toasts-specs

Guidelines that I think are nice:

- No two bars are on the screen at the same time (check out the consecutive bar animation).
- They contain at most, one action. Use a dialog if it's two or more.

I think we could set a max-width on the notices on desktop. Material guidelines encourage short strings, but they can be 2 lines if they need to be:

**Comment by rickybanister** [comment]
I actually prevented these notices from being more than one line for desktop pill-shaped situations (for now anyway). 

For the mobile width it wraps just fine and can be as long as needed.

I just realized we still need to 

- [ ] test/handle situations where there is a notice action

**Comment by mtias** [comment]
> test/handle situations where there is a notice action

I'm not sure we have these using global notices yet. Maybe @johnHackworth @artpi can remember some.

**Comment by johnHackworth** [comment]
I think there's no actions in any global notice I can think about ... but let's ping also @ebinnion to make sure the new invites flow (which if I'm not wrong use global notices) doesn't use them either

**Comment by ebinnion** [comment]
@lezama - Correct me if I'm wrong, but the only action we have in our global notices is a dismiss button, which I don't believe counts for this case.

All of the global notices we use are through the `client/state/notices/action.js` file, for which I don't see an `action` or `NoticeAction` prop.

The one place that we do use `NoticeAction` is in an inline notice: https://github.com/Automattic/wp-calypso/blob/master/client/my-sites/invites/invite-accept/index.jsx#L209

**Comment by lezama** [comment]
@ebinnion as you said, the one place that we use `NoticeAction` is in the inline notice you linked :+1:

**Comment by mtias** [comment]
Thanks for checking, folks. @rickybanister seems we can proceed with this and then look at the action once we have a case for it.

**Comment by artpi** [comment]
@mtias , @johnHackworth , @ebinnion , @lezama 
https://github.com/Automattic/wp-calypso/blob/master/client/components/global-notices/index.jsx#L109

redux global notices DO NOT accomodate button / onClick props like before.

I am gradually converting everything and we'll see if there's a true need.
IMO most of these actions are not necesarry.

My plan is to convert everything, see how many notices with actions are left and see what I can do about them.
If a prop will be needed, I will add it, but first I want a clean solution before we add too much functionality.
It is more of a janitorial decision, because code would be very simple

**Comment by mtias** [comment]
We are going to need the action button for sure, specially if we want to move some of the inline notices to global (publishing a post, etc).

**Comment by artpi** [comment]
Im aware of that, I was just waiting on the redux to settle a bit.
I can do button/href props right away
But onClick needs a bit of deliberation...
I was hoping solution will present itself :)

**Comment by mtias** [comment]
I don't think we need it now (for this PR) so no problem in waiting until we move some notice that needs it :)

**Comment by mtias** [comment]
@rickybanister rebased with master.

**Comment by shaunandrews** [comment]
Feels pretty good. I dig it on mobile.

**Comment by rickybanister** [comment]
I opened #2958 to handle some of the future enhancements mentioned in this thread (and elsewhere). I think we're pretty close to being able to merge this and improve it after.

**Comment by gwwar** [comment]
:+1: tested on Chrome, IE 11, and Edge. Code looks good and works well. This should be good to go unless there are any other design blockers.

File: client/components/global-notices/style.scss [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+		max-width: calc( 100% - 32px );
+	}
+
+	@include breakpoint( ">960px" ) {
+			top: 47px + 24px;
+			right: 24px;
+		max-width: calc( 100% - 48px );
+	}
+
+	@include breakpoint( ">1040" ) {
```
Comment (gwwar) [inline client/components/global-notices/style.scss]: I think the warning in _breakpoints.scss needs to be updated. I'm seeing:

```
WARNING: ERROR in breakpoint( >1040 ): You can only use these sizes[  480px 660px 960px 1040px ] using the following syntax [ <480px >480px 480px-660px ]
	on line 47 of assets/stylesheets/shared/mixins/_breakpoints.scss
	from line 25 of client/components/global-notices/style.scss
```

**Comment by mikeshelton1503** [comment]
Looking good! I also agree that the bottom positioning for small screens works well. :smile: 

I did find some issues and have concerns about longer notices. I'll start with the bugs:

**Long notices do not stay left aligned at all widths.** 
I cancelled a domain and noticed this:
image

**Display issues with multiple notices**
I deleted a site and apparently it took a while for the notice to display. Before it displayed I removed a another purchase and that caused both notices to display. This is probably an unlikely case but it does reveal that the notices do not stack on very wide screens and when they're side-by-side there's no padding between them.
image


------------------------------------------

Forgive my ignorance here but are all Global notices not affected by this change? I removed a purchase and still saw the current style global notice:
image

------------------------------------------

**Handling long notices**

The cancelled domain notice (mentioned above) is the only "too long" notice I have found thus far but I'm _pretty sure_ there are more in Domains and maybe Purchases too. cc @scruffian @rads @fabianapsimoes for their knowledge there.

My concern is what info will be hidden from users because of the truncation of the notices. In some cases we have relied on notices too much for relaying important details of an action. That's an issue in itself but this PR makes it relavant and more important to resolve.  I still think this PR is the right direction to go but I feel like before we get too far we should solve for long notices to avoid miscommunication with our users.

In the case of the cancelled domain notice, here's what would be cut be cut off:
image
image

I'm thinking all global notices should be shortened (and possibly messaging somewhat standardized). Then we need to find solutions for better communicating any important info currently being displayed in global notices. That could be a large undertaking though (depends on how many notices really qualify as "too long").

**Pull request merged**